### ISSUE TITLE
[Feature Request]: Ability to import multiple rules from local JSON

### ISSUE THREAD

**Issue by ericbiewener** [pqrs-org/Karabiner-Elements#3817]
I would like to be able to import multiple rules from a JSON file on my machine rather than needing to copy/paste individual rules into the "Add your own rule" text box. Another way to think about this is that the "Add predefined rule" should work with local JSON files, not just imported from the [KE-complex_modifications repo](https://github.com/pqrs-org/KE-complex_modifications).

**Comment by evan-liu** [comment]
If you'd like to give `karabiner.ts` a try, there is a [importJson()](https://evan-liu.github.io/karabiner.ts/imports/#import-json) function for that.

**Comment by ericbiewener** [comment]
Would that help? From what I can tell karabiner.ts is used to generate JSON. I'd still have to import one rule at a time in Karabiner Elements, no?

**Comment by evan-liu** [comment]
`karabiner.ts` writes to `~/.config/karabiner/karabiner.json` if [using with Node.js](https://github.com/evan-liu/karabiner.ts?tab=readme-ov-file#using-nodejs) or [Deno](https://github.com/evan-liu/karabiner.ts?tab=readme-ov-file#using-deno)

> Karabiner-Elements watches ~/.config/karabiner/karabiner.json and reloads it if updated.

Importing JSON to Karabiner-Elements is only needed when using the Online Editor. 

Sorry about the confusion and thanks for pointing it out. I will update the README.

**Comment by noptoboggan** [comment]
💯 would also like to see this implemented natively.

Though, admittedly, the main reason that I want this feature is because I periodically have to re-add all of my custom complex modifications that I have persisted as JSON files, as Karabiner-Elements clears all of my complex modifications regularly for no apparent reason.

**Comment by LumaKernel** [comment]
I agree with @noptoboggan , karabiner.json itself has some machine-specific information, so it's not suitable for persistence.  I prefer saving just pieces of configuration patches.

**Comment by LumaKernel** [comment]
I have found the hack to import complex modifications local json with terminal. Add following script to your bashrc.

```sh
# Usage: karabiner_import /path/to/keybinding.json
function karabiner_import() {
  if [ ! -f "$1" ]; then
    echo "[ERROR] File not found: $1"
    return 1
  fi
  open "karabiner://karabiner/assets/complex_modifications/import?url=file://$(realpath "$1")"
}
```

Then you can import local json files by `karabiner_import /path/to/your-local-config.json` :+1:

Thanks to genesy/karabiner-complex-rules-generator: https://github.com/genesy/karabiner-complex-rules-generator/blob/90cbbd1ae1db80bdf06910a2e1da587aada9aab7/src/components/MainForm.tsx#L227-L232

<details><summary>For Fishers</summary>
<p>

```sh
function karabiner_import --description 'Import a Karabiner-Elements complex modification JSON file. Usage: karabiner_import /path/to/config.json' -a json_path
  if test ! -f $json_path
    echo "[ERROR] File not found: $json_path"
    return 1
  end
  open "karabiner://karabiner/assets/complex_modifications/import?url=file://"(realpath $json_path)
end
```

</p>
</details>

**Comment by peterjc** [comment]
I'm using the trick suggested (actual filename omitted):

```console
$ open "karabiner://karabiner/assets/complex_modifications/import?url=file:///Users/peterjc/XXX/XXX.json"
```

This works, in that Karabiner Elements Settings pops up a dialogue with:

```
Import file from file:///Users/peterjc/XXX/XXX.json

(title string for start of my JSON file)
-------------------------------------------------------------
(description of my first rule)
(description of my second rule)
...
(description of my last rule)

[cancel] [import]
```

I click on the import button, then have a new popup:

```
Import more rules from the Internet (Open a web browser)

[search box]

Examples
Change caps_lock ... [enable]
Change right_command... [enable]
Change spacebar ... [enable]

                       [Enable All]
```

There is no sign of *my* new rules here, as though the import silently failed. The search box doesn't reveal any matches (although the three examples vanish).

Looking at the log within the settings app, I can see nothing about importing this file.

I have also tried copying my file to ``~/.config/karabiner/assets/complex_modifications/XXX.json`` but again, nothing seems to have happened with after restarting the app.

Am I doing something wrong? This is KE version 15.3.0 (current latest).
